src/proposals/20250109/SparkEthereum_20250109.t.sol

```diff
@@ -0,0 +1,926 @@
+
+        deal(Ethereum.USDS, Ethereum.ALM_PROXY, 50_000_000e18);
+        // Use deal2 for USDC because storage is not set in a common way
+        deal2(Ethereum.USDC, Ethereum.ALM_PROXY, 50_000_000e6);
+
+        // USDS
+
+        vm.startPrank(Ethereum.ALM_RELAYER);
+
+        bytes32 usdsDepositKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_DEPOSIT(),
+            address(ausds)
+        );
+        bytes32 usdsWithdrawKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_WITHDRAW(),
+            address(ausds)
+        );
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsDepositKey), 50_000_000e18);
+        assertEq(usds.balanceOf(Ethereum.ALM_PROXY),             50_000_000e18);
+        assertEq(ausds.balanceOf(Ethereum.ALM_PROXY),            0);
+
+        controller.depositAave(address(ausds), 50_000_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsDepositKey), 0);
+        assertEq(usds.balanceOf(Ethereum.ALM_PROXY),             0);
+        assertEq(ausds.balanceOf(Ethereum.ALM_PROXY),            50_000_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsWithdrawKey), type(uint256).max);
+
+        // USDC
+
+        bytes32 usdcDepositKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_DEPOSIT(),
+            address(ausdc)
+        );
+        bytes32 usdcWithdrawKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_WITHDRAW(),
+            address(ausdc)
+        );
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 50_000_000e6);
+        assertEq(usdc.balanceOf(Ethereum.ALM_PROXY),             50_000_000e6);
+        assertEq(ausdc.balanceOf(Ethereum.ALM_PROXY),            0);
+
+        controller.depositAave(address(ausdc), 50_000_000e6);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 0);
+        assertEq(usdc.balanceOf(Ethereum.ALM_PROXY),             0);
+        assertEq(ausdc.balanceOf(Ethereum.ALM_PROXY),            50_000_000e6);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcWithdrawKey), type(uint256).max);
+
+        // All limits should be reset in 2 days + 1 (rounding)
+        skip(2 days + 1);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsDepositKey), 50_000_000e18);
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 50_000_000e6);
+    }
+
+    function test_BASE_ControllerUpgrade() public onChain(ChainIdUtils.Base()) {
+        // Deployment configuration is checked inside the spell
+
+        IALMProxy proxy = IALMProxy(Base.ALM_PROXY);
+
+        assertEq(proxy.hasRole(proxy.CONTROLLER(), Base.ALM_CONTROLLER),     true);
+        assertEq(proxy.hasRole(proxy.CONTROLLER(), BASE_NEW_ALM_CONTROLLER), false);
+
+        executeAllPayloadsAndBridges();
+
+        assertEq(proxy.hasRole(proxy.CONTROLLER(), Base.ALM_CONTROLLER),     false);
+        assertEq(proxy.hasRole(proxy.CONTROLLER(), BASE_NEW_ALM_CONTROLLER), true);
```

Comment:
Same updates needed here, need to assert relayer and freezer roles are set in the controller, as well as controller role transfer in rate limits

Reply:
added